db/snapshot/backup_task.hh

```diff
@@ -52,13 +55,17 @@ class backup_task_impl : public tasks::task_manager::task::impl {
     // the concurrency level of S3 uploads
     constexpr static size_t max_concurrency = 32 << 10;
     semaphore _concurrency_sem{max_concurrency};
+    shard_id _backup_shard;
+    sharded<foreign_ptr<shared_ptr<sstables::manager_signal_connection_type>>> _subscriptions;
```

Comment:
Why foreign_ptr? Doesn't sharded<whatever> handle the allocation to shard affinity naturally?

Reply:
dropped in v3 ([7ab3391](https://github.com/scylladb/scylladb/commit/7ab3391f15e68adfff6187385be438b96bb3d57e)), in favor of `sharded<sstables_manager_for_table>`